packages/query/src/stacks/stx20/stx20-tokens.hooks.ts

```diff
@@ -11,9 +11,11 @@ import { createGetStx20BalancesQueryOptions } from './stx20-tokens.query';
 
 function createStx20CryptoAssetInfo(stx20Balance: Stx20Balance): Stx20CryptoAssetInfo {
   return {
+    assetCategory: 'fungible',
+    assetType: 'stx20',
     name: 'stx-20',
     symbol: stx20Balance.ticker,
-  };
+  } as Stx20CryptoAssetInfo;
```

Comment:
```suggestion
  } satisfies Stx20CryptoAssetInfo;
```

?

Reply:
although why is this cast needed in first place when return type is declared

Reply:
This type assertion is needed to ignore the shared fungible base fields `hasMemo` and `decimals`. `Stx20CryptoAssetInfo` was previously not extending any base types, so this is a workaround to get it into the heirarchy. As a follow-up issue we should try to extract the shared fields to more specific types as I suspect only a couple of assets actually use them.

Reply:
this can go with your latest changes though right as you removed from the base class?

Reply:
I didn't remove `decimals` and `hasMemo`, I just moved them from the base asset to the base fungible asset. It existed previously on 6 of 7 fungible assets, with this as the lone exception (`Stx20CryptoAssetInfo`), so I figured a temporary type assertion was preferable than redefining these fields 6 times on each fungible model type. But I think this definitely warrants a follow-up issue to figure out on which asset types exactly these fields are relevant/required and make them much more specific.

Reply:
Had a look into this. There's actually no need to satisfy the return type of `Stx20CryptoAssetInfo`.

Typing something as X when it isn't actually X, is a crime demanding the most severe punishment lol. It defeats the point of having rich type information if you can't trust it.

Nothing is expecting the full `Stx20CryptoAssetInfo` here, so typing this function as returning `Partial<Stx20CryptoAssetInfo>` resp is totally fine. 

```suggestion
  };
```